examples/default/Makefile

```diff
@@ -37,14 +37,21 @@ USEMODULE += ps
 # include and auto-initialize all available sensors
 USEMODULE += saul_default
 
-BOARD_PROVIDES_NETIF := acd52832 airfy-beacon b-l072z-lrwan1 cc2538dk fox \
-        iotlab-m3 iotlab-a8-m3 lsn50 mulle microbit native nrf51dk \
-        nrf51dongle nrf52dk nrf52840dk nrf52840-mdk nrf6310 \
-        openmote-cc2538 pba-d-01-kw2x remote-pa remote-reva samr21-xpro \
-        spark-core telosb yunjia-nrf51822 z1
+BOARD_PROVIDES_NETIF := acd52832 airfy-beacon b-l072z-lrwan1 calliope-mini \
+                        cc2538dk esp32-mh-et-live-minikit esp32-olimex-evb \
+                        esp32-wemos-lolin-d32-pro esp32-wroom-32 \
+                        esp32-wrover-kit firefly fox i-nucleo-lrwan1 \
+                        iotlab-a8-m3 iotlab-m3 lsn50 microbit mulle native \
+                        nrf51dk nrf51dongle nrf52832-mdk nrf52840-mdk \
+                        nrf52840dk nrf52dk nrf6310 openmote-cc2538 \
+                        particle-argon particle-boron particle-xenon \
+                        pba-d-01-kw2x reel remote-pa remote-reva remote-revb \
+                        ruuvitag samr21-xpro samr30-xpro telosb thingy52 \
+                        yunjia-nrf51822 z1
 
 ifneq (,$(filter $(BOARD),$(BOARD_PROVIDES_NETIF)))
   # Use modules for networking
+  FEATURES_REQUIRED += netif
```

Comment:
Is not the 'FEATURES_REQUIRED' supposed to be in a dependency of one of the used module ?

Reply:
Maybe but which one ? ;)

I wanted to put this as a dependency of `gnrc_netif` or `netdev_default` but the problem then is that it will be difficult to build an application for a board where one wires manually an external radio. That's why I left it at the application level.

Reply:
> Maybe but which one ? ;)

I would say `netdev_default` as it is the one that pulls the radio in `boards/board/Makefile.dep` but maybe `gnrc_netif` is better suited. @miri64 knows better than me.

> I wanted to put this as a dependency of gnrc_netif or netdev_default but the problem then is that it will be difficult to build an application for a board where one wires manually an external radio. That's why I left it at the application level.

If you wire an external radio, it is for me a "different" board. And you need to write you external radio configuration too
```
boards/hamilton/include/board.h: * @name AT86RF233 configuration
boards/hamilton/include/board.h:#define AT86RF2XX_PARAM_SPI      SPI_DEV(0)
boards/hamilton/include/board.h:#define AT86RF2XX_PARAM_SPI_CLK  SPI_CLK_5MHZ
boards/hamilton/include/board.h:#define AT86RF2XX_PARAM_CS       GPIO_PIN(PB, 31)
boards/hamilton/include/board.h:#define AT86RF2XX_PARAM_INT      GPIO_PIN(PB, 0)
boards/hamilton/include/board.h:#define AT86RF2XX_PARAM_SLEEP    GPIO_PIN(PA, 20)
boards/hamilton/include/board.h:#define AT86RF2XX_PARAM_RESET    GPIO_PIN(PB, 15)
```

So would need to have some files adapted.
However, for me the re-use of a board or modifying a board is its own topic. Something I would like but has lots of dependencies https://github.com/RIOT-OS/RIOT/issues/11674


Is it possible to "modify" a board from an application Makefile ? not entirely as it does not define `Makefile.include` and `Makefile.dep` that are parsed with the others.

That last point could be addressed. To include `$(APPDIR)/Makefile.features`, `$(APPDIR)/Makefile.include` and `$(APPDIR)/Makefile.dep` in they exist.
However it would not currently entirely work currently due to the hacks:

https://github.com/RIOT-OS/RIOT/blob/761530caf50e94fa5a50fa2f11f3b3ce977b9ee0/boards/common/nrf52/Makefile.include#L12-L23

The hacks will be able to be removed after https://github.com/RIOT-OS/RIOT/issues/9913 if we can move forward with https://github.com/RIOT-OS/RIOT/pull/11478